File: 5-Matematica/1279-Ano-Bissexto-ou-Ano-nao-Bissexto/1279.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <ctype.h>

int main() {
    char *year = (char*)malloc(2001 * sizeof(char));
    bool pres = false, test, test2;
    while ((fgets(year, 2001, stdin) != NULL)) {
        if (pres) {
            printf ("\n");
        }
        else {
            pres = true;
        }
        int cont[7] = {0}; // ppar, pimpar, sn, uni, dez, cent, centmil
        for (int i = 0; i <= 2000; ++i) {
            if (isdigit(year[i]) != 0) {
                if ((i + 1) % 2 == 0) {
                    cont[0] += year[i] - 48;
                }
                else {
                    cont[1] += year[i] - 48;
                }
                cont[2] += year[i] - 48;
            }
            else {
                cont[3] = (year[i - 1] - 48);
                cont[4] = (year[i - 2] - 48);
                cont[5] = (year[i - 3] - 48);
                cont[6] = (year[i - 4] - 48);
                break;
            }
        }
        test = false, test2 = false;
        if (((10 * cont[4] + cont[3]) % 4 == 0 && (cont[3] != 0 || cont[4] != 0)) || (cont[3] == 0 && cont[4] == 0 && (((10 * cont[6] + cont[5]) * 100) % 400 == 0))) {
            printf ("This is leap year.\n");
            test = true;
        }
        if ((cont[3] == 5 || cont[3] == 0) && (cont[2] % 3 == 0)) {
            printf ("This is huluculu festival year.\n");
            test2 = true;
        }
        if (test == true && (cont[3] == 5 || cont[3] == 0) && ((cont[0] - cont[1]) % 11 == 0 || cont[0] - cont[1] == 0)) {
            printf ("This is bulukulu festival year.\n");
        }
        else if (test == false && test2 == false) {
            printf ("This is an ordinary year.\n");
        }
    }
    return 0;
}

File: 5-Matematica/1279-Ano-Bissexto-ou-Ano-nao-Bissexto/1279.py
pres = False
store = []
while (1):
    try:
        store.append(list(map(int, input())))
    except EOFError:
        for number in store:
            if pres:
                print ()
            pres = True
            isleap, ishuluculu = False, False
            ppar, pimpar = 0, 0
            
            for index, i in enumerate (number):
                if index % 2 == 0:
                    ppar += i
                else:
                    pimpar += i
            if ((number[-2]*10 + number[-1]) % 4 == 0 and (number[-2] != 0 or number[-1] != 0)) or ((number[-2] == 0 and number[-1] == 0) and ((number[-4]*10 + number[-3]) * 100) % 400 == 0):
                print ("This is leap year.")
                isleap = True
            if (((ppar + pimpar) % 3 == 0) and (number[-1] == 0 or number[-1] == 5)):
                print ("This is huluculu festival year.")
                ishuluculu = True
            if ((ppar - pimpar) % 11 == 0 and isleap):
                print ("This is bulukulu festival year.")
            elif not isleap and not ishuluculu:
                print ("This is an ordinary year.")
        break